inference/generativeai/llm-workshop/lab4-openchatkit/deploy_openchatkit_on_sagemaker_deepspeed.ipynb

```diff
@@ -0,0 +1,1141 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c9ff1d6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "prompts = [\n",
+    "    \"<human> Classify the sentiment of the following sentence into Positive, Neutral, or Negative: How about the following sentence: It is raining outside and I feel so blue <bot>:\"\n",
+    "]\n",
+    "response_model = smr_client.invoke_endpoint(\n",
+    "    EndpointName=endpoint_name,\n",
+    "    Body=json.dumps(\n",
+    "        {\n",
+    "            \"inputs\": prompts,\n",
+    "            \"parameters\": {\n",
+    "                \"temperature\": 0.6,\n",
+    "                \"top_k\": 40,\n",
+    "            },\n",
+    "        }\n",
+    "    ),\n",
+    "    ContentType=\"application/json\",\n",
+    ")\n",
+    "\n",
+    "response_model[\"Body\"].read().decode(\"utf8\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c27db166",
+   "metadata": {},
+   "source": [
+    "## Extending the chatbot with retrieval system for live-updating answers and Moderation\n",
+    "\n",
+    "In this example, we demonstrated how to use SageMaker large model inference container to host OpenChatKit. We used DeepSpeed’s model parallel techniques with multiple GPUs on a single SageMaker machine learning instance. OpenChatKit also includes an extensible retrieval system. With the retrieval system the chatbot is able to incorporate regularly updated or custom content, such as knowledge from Wikipedia, news feeds, or sports scores in response. The additional component of OpenChatKit is a 6 billion parameter moderation model fine-tuned from GPT-JT. In chat applications, the moderation model runs in tandem with the main chat model, checking the user utterance for any inappropriate content. Based on the moderation model’s assessment, the chatbot can limit the input to moderated subjects. For more narrow tasks the moderation model can be used to detect out-of-domain questions and override when the question is not on topic Please refer to [this](https://www.together.xyz/blog/openchatkit) blog post to extend this model with retrieval system.\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b11c344c",
+   "metadata": {},
+   "source": [
+    "## Clean Up"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9947d080",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# - Delete the end point\n",
+    "sm_client.delete_endpoint(EndpointName=endpoint_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f92e2391",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# - In case the end point failed we still want to delete the model\n",
+    "sm_client.delete_endpoint_config(EndpointConfigName=endpoint_config_name)\n",
```

Comment:
https://docs.djl.ai/docs/demos/aws/sagemaker/large-model-inference/standard_template_with_LMI_solution.html#clean-up-the-environment Suggest to use pysdk

Reply:
Great suggestion, however, PySDK is not up to speed with having new and latest features in SageMaker so I prefer to keep it using boto3 until we have full feature parity.  Additionally, would like to keep it more friendly to MLOps so that they can automate this in their CICD pipeline using light weight boto3 as compared to more bulkier PySDK.